src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+
 public class SalaryInfo {
+    private final SimpleDateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        try {
+            Date fromDate = dateFormat.parse(dateFrom);
+            Date toDate = dateFormat.parse(dateTo);
+            HashMap<String, Integer> salaryMap = new HashMap<>();
+            for (String name : names) {
+                salaryMap.put(name, 0);
+            }
```

Comment:
we can do without this cycle, i.e. when we want to put a value in the map, you can first check if there is such a key there and if there is, then update its value, if not put